lib/backend.js

```diff
@@ -199,22 +249,40 @@ Backend.prototype._sanitizeSnapshot = function(agent, projection, collection, id
     try {
       projections.projectSnapshot(projection.fields, snapshot);
     } catch (err) {
-      return callback(err);
+      return process.nextTick(function() {
```

Comment:
If we chance this to be async, I think it is important that the callback below is async, too. Trigger itself is sync, so the call could be sync or async depending on the middleware that people write. Probably safer to make it all async, but let's make sure that we cover all cases if we are going to make a change.

Also, it might be better to make the sync callback change as a separate sharedb release just in case it has an impact on others' code. I'd like to do this, but I think it may be best to do separately.